src/coreclr/nativeaot/System.Private.StackTraceMetadata/src/Internal/StackTraceMetadata/MethodNameFormatter.cs

```diff
@@ -181,6 +181,39 @@ private void EmitMethodParameters(MethodSignature methodSignature)
             _outputBuilder.Append(')');
         }
 
+        /// <summary>
+        /// Emit parenthesized method argument type list with parameter names.
+        /// </summary>
+        /// <param name="methodHandle">Method handle to use for parameter formatting</param>
+        private void EmitMethodParameters(MethodHandle methodHandle)
+        {
+            Method method = methodHandle.GetMethod(_metadataReader);
+            HandleCollection typeVector = method.Signature.GetMethodSignature(_metadataReader).Parameters;
+            ParameterHandleCollection.Enumerator names = method.Parameters.GetEnumerator();
+
+            bool first = true;
+
+            _outputBuilder.Append('(');
+            foreach (Handle handle in typeVector)
+            {
+                if (first)
+                {
+                    first = false;
+                }
+                else
+                {
+                    _outputBuilder.Append(", ");
+                }
+
+                names.MoveNext();
+
+                EmitTypeName(handle, namespaceQualified: false);
+                _outputBuilder.Append(' ');
+                EmitString(names.Current.GetParameter(_metadataReader).Name);
```

Comment:
The name could be an empty string - let's skip adding the space and name if that's the case.